sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -402,6 +425,109 @@ def _get_environment(
         env = self._environment_from_row(row)
         return env
 
+    @transactional()
+    def add_interval(
+        self,
+        snapshot: Snapshot,
+        start: TimeLike,
+        end: TimeLike,
+        is_dev: bool = False,
+    ) -> None:
+        logger.info("Adding interval for snapshot %s", snapshot.snapshot_id)
+
+        self.engine_adapter.insert_append(
+            self.intervals_table,
+            _intervals_to_df([snapshot], start, end, is_dev, False),
+            columns_to_types=self._interval_columns_to_types,
+        )
+
+    @transactional()
+    def remove_interval(
+        self,
+        snapshots: t.Iterable[SnapshotInfoLike],
+        start: TimeLike,
+        end: TimeLike,
+        all_snapshots: t.Optional[t.Iterable[Snapshot]] = None,
+    ) -> None:
+        all_snapshots = all_snapshots or self._get_snapshots_with_same_version(snapshots)
+
+        if logger.isEnabledFor(logging.INFO):
```

Comment:
why do you need to check this?

Reply:
To avoid constructing a string that used only for logging purposes if the logging is not even enabled.

Reply:
Oops, this looks like a leftover artifact.

Reply:
ah, no, still relevant

Reply:
meh i think it's ok to just always generate it